python/hyperon/atoms.py

```diff
@@ -187,4 +189,54 @@ def ValueAtom(value, type_name=None, atom_id=None):
     return G(ValueObject(value, atom_id), _type_sugar(type_name))
 
 def MatchableAtom(value, type_name=None, atom_id=None):
-    return G(MatchableObject(value, atom_id), _type_sugar(type_name))
\ No newline at end of file
+    return G(MatchableObject(value, atom_id), _type_sugar(type_name))
+
+
+class Bindings:
+
+    def __init__(self, bindings: hp.CBindings | None = None):
+        if bindings is None:
+            self.cbindings = hp.bindings_new()
+        else:
+            self.cbindings = bindings
+
+    def __del__(self):
+        if self.cbindings is not None:
+            hp.bindings_free(self.cbindings)
+
+    def __eq__(self, other):
+        return (isinstance(other, Bindings) and
+                hp.bindings_eq(self.cbindings, other.cbindings))
+
+    def __repr__(self):
+        return hp.bindings_to_str(self.cbindings)
+
+    def __deepcopy__(self, memodict={}):
+        return self.clone()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.cbindings is not None:
+            hp.bindings_free(self.cbindings)
+            self.cbindings = None
+
+    def clone(self):
+        return Bindings(hp.bindings_clone(self.cbindings))
+
+    @staticmethod
+    def merge(left, right):
+        return Bindings(hp.bindings_merge(left.cbindings, right.cbindings))
+
+    def add_var_bindings(self, var_name:str, atom: Atom) -> bool:
+        return hp.bindings_add_var_bindings(self.cbindings, var_name, atom.catom)
+
+    def is_empty(self) -> bool:
+        return hp.bindings_is_empty(self.cbindings)
+
+    def resolve(self, var_name: str) -> Atom:
+        return Atom(hp.bindings_resolve(self.cbindings, var_name))
```

Comment:
One more thing, the `Atom._from_catom` factory method should be used to correctly create new atom from C structure here and below.